mllib/src/main/scala/org/apache/spark/ml/tree/impl/BaggedPoint.scala

```diff
@@ -82,16 +92,16 @@ private[spark] object BaggedPoint {
       val rng = new XORShiftRandom
       rng.setSeed(seed + partitionIndex + 1)
       instances.map { instance =>
-        val subsampleWeights = new Array[Double](numSubsamples)
+        val subsampleCounts = new Array[Int](numSubsamples)
         var subsampleIndex = 0
         while (subsampleIndex < numSubsamples) {
           val x = rng.nextDouble()
-          subsampleWeights(subsampleIndex) = {
-            if (x < subsamplingRate) 1.0 else 0.0
+          subsampleCounts(subsampleIndex) = {
```

Comment:
The while body could be a little tighter as:
```
if (rng.nextDouble() < subsamplingRate) {
  subsampleCounts(subsampleIndex) = 1
}
subsampleIndex += 1
```

Reply:
good suggestion, done!